pkg/oc/cli/util/clientcmd/gating.go

```diff
@@ -1,126 +0,0 @@
-package clientcmd
-
-import (
-	"fmt"
-
-	"k8s.io/apimachinery/pkg/api/errors"
-	"k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime/schema"
-	"k8s.io/client-go/discovery"
-
-	authorization "github.com/openshift/api/authorization/v1"
-)
-
-// LegacyPolicyResourceGate returns err if the server does not support the set of legacy policy objects (< 3.7)
-func LegacyPolicyResourceGate(client discovery.ServerResourcesInterface) error {
-	// The server must support the 4 legacy policy objects in either of the GV schemes.
-	_, all, err := DiscoverGroupVersionResources(client,
-		schema.GroupVersionResource{
-			Group:    authorization.LegacySchemeGroupVersion.Group,
-			Version:  authorization.LegacySchemeGroupVersion.Version,
-			Resource: "clusterpolicies",
-		},
-		schema.GroupVersionResource{
-			Group:    authorization.LegacySchemeGroupVersion.Group,
-			Version:  authorization.LegacySchemeGroupVersion.Version,
-			Resource: "clusterpolicybindings",
-		},
-		schema.GroupVersionResource{
-			Group:    authorization.LegacySchemeGroupVersion.Group,
-			Version:  authorization.LegacySchemeGroupVersion.Version,
-			Resource: "policies",
-		},
-		schema.GroupVersionResource{
-			Group:    authorization.LegacySchemeGroupVersion.Group,
-			Version:  authorization.LegacySchemeGroupVersion.Version,
-			Resource: "policybindings",
-		},
-	)
-
-	if err != nil {
-		return err
-	}
-	if all {
-		return nil
-	}
-	_, all, err = DiscoverGroupVersionResources(client,
-		schema.GroupVersionResource{
-			Group:    authorization.SchemeGroupVersion.Group,
-			Version:  authorization.SchemeGroupVersion.Version,
-			Resource: "clusterpolicies",
-		},
-		schema.GroupVersionResource{
-			Group:    authorization.SchemeGroupVersion.Group,
-			Version:  authorization.SchemeGroupVersion.Version,
-			Resource: "clusterpolicybindings",
-		},
-		schema.GroupVersionResource{
-			Group:    authorization.SchemeGroupVersion.Group,
-			Version:  authorization.SchemeGroupVersion.Version,
-			Resource: "policies",
-		},
-		schema.GroupVersionResource{
-			Group:    authorization.SchemeGroupVersion.Group,
-			Version:  authorization.SchemeGroupVersion.Version,
-			Resource: "policybindings",
-		},
-	)
-
-	if err != nil {
-		return err
-	}
-	if all {
-		return nil
-	}
-
-	return fmt.Errorf("the server does not support legacy policy resources")
-}
-
-// DiscoverGroupVersionResources performs a server resource discovery for each filterGVR, returning a slice of
-// GVRs for the matching Resources, and a bool for "all" indicating that each item in filterGVR was found.
-func DiscoverGroupVersionResources(client discovery.ServerResourcesInterface, filterGVR ...schema.GroupVersionResource) ([]schema.GroupVersionResource, bool, error) {
```

Comment:
Hmm we may need this in the future?

Reply:
> Hmm we may need this in the future?

And we can revisit at that time.  Git keeps the "might come in handy's" without us keeping cruft.